src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -1894,5 +1976,680 @@ struct S
+                System.Console.Write(3);
+                break;
+        }
+    }
+}";
+            var comp = CreateCompilationWithIndexAndRange(src, parseOptions: TestOptions.RegularWithListPatterns);
+            comp.VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "2");
+        }
+
+        [Fact]
+        public void Subsumption_10()
+        {
+            var src = @"
+class C
+{
+    public int X { get; }
+    public int Y { get; }
+    public C F { get; }
+    static void Test(C[] a) 
+    {
+        switch (a)
+        {
+            case [.., {X:> 0, Y:0}]:
+            case [{Y:0, X:> 0}]:
+                break;
+        }
+        switch (a)
+        {
+            case [.., {X:> 0, F.Y: 0}]:
+            case [..[{F.Y: 0, X:> 0}]]:
+                break;
+        }
+    }
+}
+";
+            var comp = CreateCompilationWithIndexAndRange(src, parseOptions: TestOptions.RegularWithListPatterns);
+            comp.VerifyEmitDiagnostics(
+                // (12,18): error CS8120: The switch case is unreachable. It has already been handled by a previous case or it is impossible to match.
+                //             case [{Y:0, X:> 0}]:
+                Diagnostic(ErrorCode.ERR_SwitchCaseSubsumed, "[{Y:0, X:> 0}]").WithLocation(12, 18),
+                // (18,18): error CS8120: The switch case is unreachable. It has already been handled by a previous case or it is impossible to match.
+                //             case [..[{F.Y: 0, X:> 0}]]:
+                Diagnostic(ErrorCode.ERR_SwitchCaseSubsumed, "[..[{F.Y: 0, X:> 0}]]").WithLocation(18, 18)
+                );
+        }
+
+        [Fact]
+        public void Subsumption_11()
+        {
+            var src = @"
+class C
+{
+    public int X { get; }
+    public int Y { get; }
+
+    public void Deconstruct(out C c1, out C c2) => throw null;
+
+    static void Test(C[] a) 
+    {
+        switch (a)
+        {
+            case [.., (_, { X: 0 })]:
+            case [({ X: 0 }, _)]: // ok
+                break;
+        }
+        switch (a)
+        {
+            case [.., (_, { X: 0 })]:
+            case [(_, { X: 0 })]: // err
+                break;
+        }
+    }
+}
+";
+            var comp = CreateCompilationWithIndexAndRange(src, parseOptions: TestOptions.RegularWithListPatterns);
+            comp.VerifyEmitDiagnostics(
+                // (20,18): error CS8120: The switch case is unreachable. It has already been handled by a previous case or it is impossible to match.
+                //             case [(_, { X: 0 })]: // err
+                Diagnostic(ErrorCode.ERR_SwitchCaseSubsumed, "[(_, { X: 0 })]").WithLocation(20, 18));
+        }
+
+        [Fact]
+        public void Subsumption_12()
+        {
+            var src = @"
+class C
+{
+    void Test(int[] a)
+    {
+        _ = a switch
+        {
+            { Length: not 1 }  => 0,
+            [<0, ..] => 0,
+            [..[>= 0]] or [..null] => 1,
```

Comment:
Why is there no error on the `[..null]` bit? Shouldn't that _never_ match?

Reply:
> Shouldn't that never match?

An array slice is still an array so null is considered a possible value.

Reply:
Ohhh, _slice_. I think I got confused after a few hours of reading tests and forgot this was a slice. Thanks!

Reply:
Btw, as FYI for @333fred, there's a work item tracking subsumption analysis with nullability, but that code is elsewhere and will be addressed separately.